### ISSUE TITLE
"target_link_directories" error on CMake older than 3.13

### ISSUE THREAD

**Issue by Wohlstand** [libsdl-org/SDL#5096]
Hello!

I do see that the minimal version of CMake 3.0.0 was specified, but since it uses the "target_link_directories", de-facto the minimal CMake version is 3.13: https://cmake.org/cmake/help/latest/command/target_link_directories.html

```
CMake Error at cmake/sdlchecks.cmake:649 (target_link_directories):
  Unknown CMake command "target_link_directories".
Call Stack (most recent call first):
  CMakeLists.txt:1262 (CheckWayland)
```

Could you change the minimal version into 3.13, or avoid the target_link_directories usage on CMake older than 3.13?

---
I do have the SDL 2.0.18 stable version

**Comment by glebm** [comment]
I think it's fine to bump the minimum to 3.13 at this point.

**Comment by icculus** [comment]
3.13 was released in November 2018. We just rolled back to 3.0.0 because we felt 3.11 was too new, fwiw.

**Comment by glebm** [comment]
What kind of distro would have CMake 3.0.0 with SDL 2.0.20? What use case am I not thinking of?

**Comment by Wohlstand** [comment]
Ubuntu 18 has CMake older than 3.13

**Comment by sezero** [comment]
Are there really no alternatives to `target_link_directories` instead of bumping our requirement to 3.13+?

**Comment by Wohlstand** [comment]
> Are there really no alternatives to `target_link_directories` instead of bumping our requirement to 3.13+?

I guess, the global `link_directories()` only.

If the thing makes the build of only one thing (the library itself), doesn't build utilities for external things, etc. You are absolutely fine to use the link_directories() easily. Don't worry about sub-directory use, because all changes link_directories() and other global CMane environment changes, gets affected only sub-directory itself and only other sub-directories which was defined at THIS project itself. The only real reason to use the target_link_directories is if you going to build multiple executables and modules that should link DIFFERENT libraries at different places (even, for example, build the same module for different architectures using the same project, or something also).

**Comment by icculus** [comment]
Can we polyfill this with some sort of:

```cmake
if(CMakeVersion < 3.13)
  macro(target_link_directory X)
    link_directories(X)
   endmacro()
endif()
```

(With proper syntax, or course.)

**Comment by Wohlstand** [comment]
Should work, would try some hand-made unit test, but, tomorrow, as it's almost 5 AM at my time zone and I am tired.

**Comment by glebm** [comment]
> Ubuntu 18 has CMake older than 3.13

Ubuntu 18 has SDL 2.0.8 and is unlikely to ever package a newer version.

**Comment by glebm** [comment]
@Wohlstand 
>  because all changes link_directories() and other global CMake environment changes, gets affected only sub-directory itself 

I didn't realize  `link_directories` only affects subdirectories. That would make `link_directories` OK in my book as it wouldn't break the FetchContent/add_subdirectory use-case.

**Comment by icculus** [comment]
> Ubuntu 18 has SDL 2.0.8 and is unlikely to ever package a newer version.

Which is why people will build from source code.

This isn't up for debate, we aren't bumping the CMake version right now.

**Comment by Wohlstand** [comment]
Btw, why assigned to 2.0.22? In fact, it's a bug that makes an unexpected error on older CMakes, and it needs a quick fix to get it to work. Maybe I'll try to trick some by myself and send the pull request to close this quicker?

**Comment by icculus** [comment]
We're shipping 2.0.20 as a quick fix release due to a showstopper in 2.0.18, so we moved _everything_ to 2.0.22 by default...but if you want a patch included for this issue, I will take it for .20.

Tentatively moving this back into the 2.0.20 milestone for now.
